app/Http/Controllers/Item/ItemController.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Controllers\Controller;
+use App\Http\Controllers\Item\ItemRepository;
+use App\Item;
+use app\Http\Requests\CreateItemRequest;
+
+class ItemController extends Controller
+{
+    private $itemRepository;
+
+    public function __construct(ItemRepository $itemRepository)
+    {
+        $this->itemRepository = $itemRepository;
+    }
+
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        // TO-DO
+        // Ask fro help from Jeseph of how to use the config values on pagination
+        // get the pagination number or a default
+        $items = $this->$itemRepository ?? 10;
+        return view('items' . config('settings.pagination.small') . '.index')->with(['$items' => $items]);
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        return view('items.create');
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  CreateItemRequest  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(CreateItemRequest $request)
+    {
+        $data = $request->all();
+        $data['name'] = strtoupper($data['size'] . "' " . $data['code'] . ' ' . $data['brand']);
+        Item::create($data);
+        return redirect()->to('items')->withSuccess('Item created Successfully');
```

Comment:
You would have less headache if you used route names instead of urls. The headache comes in when you are told to change the url from say, `items` to `inventory-items`. So instead of `redirect()->to('items')`, you would use `redirect()->route('items.index')`. In your route file, if using a resource controller, then you would simply define as `Route::resource('items', 'Item\ItemController');` or without resource controller as `Route::get('items', 'Item\ItemController')->name('items.index');`

Reply:
Wow, I'll be using resource controller